src/lib/multicall/utils.ts

```diff
@@ -0,0 +1,131 @@
+import { Web3Provider } from "@ethersproject/providers";
+import { CHAIN_NAMES_MAP, getRpcUrl } from "config/chains";
+import { ContractCallContext, Multicall } from "ethereum-multicall";
+import { CallContext, ContractCallResults } from "ethereum-multicall/dist/esm/models";
+import { ethers } from "ethers";
+import { bigNumberify } from "lib/numbers";
+import { getFallbackProvider } from "lib/rpc";
+import { sleep } from "lib/sleep";
+import { MulticallRequestConfig, MulticallResult } from "./types";
+
+export const MAX_TIMEOUT = 2000;
+
+export async function executeMulticall(
+  chainId: number,
+  library: Web3Provider | undefined,
+  request: MulticallRequestConfig<any>
+) {
+  // Try to use rpc provider of the connected wallet
+  let provider = library ? library.getSigner().provider : undefined;
+
+  // Wait for initialization to chech the network
+  await provider?.ready;
+
+  // If the wallet is not connected or the network does not match the chainId of the request, create a new rpc provider
+  if (!provider || provider.network?.chainId !== chainId) {
+    const rpcUrl = getRpcUrl(chainId);
+
+    provider = new ethers.providers.StaticJsonRpcProvider(rpcUrl, { chainId, name: CHAIN_NAMES_MAP[chainId] });
+  }
+
+  const multicall = getMulticallLib(provider as ethers.providers.JsonRpcProvider);
+
+  const formattedReq = formatMulticallRequest(request);
+
+  const requestPromise = Promise.race([
+    multicall.call(formattedReq),
+    sleep(MAX_TIMEOUT).then(() => Promise.reject("rpc timeout")),
+  ]).catch((e) => {
+    // eslint-disable-next-line no-console
+    console.error("multicall error:", e, request);
```

Comment:
error is also logged here https://github.com/gmx-io/gmx-interface/pull/381/files#diff-85c30e0bd513da2abf50f365526b59ef071000f8712cbc5c781441ede538d6eaR56

will be it enough to log it in one place only?